crates/ruff_linter/src/registry/rule_set.rs

```diff
@@ -243,6 +244,20 @@ impl RuleSet {
         self.0[index] & mask != 0
     }
 
+    /// Returns `true` if any of the rules in `rules` are in this set.
+    #[inline]
+    pub const fn any(&self, rules: &[Rule]) -> bool {
+        let mut any = false;
+        let mut i = 0;
+
+        while i < rules.len() {
```

Comment:
It looks like this entire function could just be written as `rules.iter().any(|r| self.contains(r))`? Did you try that and it was slower? (I believe it also has the benefit of short circuiting, which may or may not help depending on the typical length of `rules`.)

Reply:
The typical length is like 1-8 rules (where 8 is rare). The downside is that the function can't be const anymore ;)

Reply:
Ah I missed the `const` requirement.